src/Common/filesystem.h

```diff
@@ -26,9 +29,32 @@ namespace filesystem = std::filesystem;
 inline bool directoryExists(seissol::filesystem::directory_entry entry) {
   return entry.exists();
 }
-}
+} // namespace seissol
 
 #endif // EXPERIMENTAL_FS
 
+
+namespace seissol {
+inline void generateBackupFileIfNecessary(std::string fileName,
+                                          std::string fileExtension,
+                                          std::optional<std::string> timeStamp = {}) {
+  std::stringstream fullName;
+  fullName << fileName << '.' << fileExtension;
+  seissol::filesystem::path path(fullName.str());
+  seissol::filesystem::directory_entry entry(path);
+
+  if (seissol::directoryExists(entry)) {
+    if (!timeStamp.has_value()) {
```

Comment:
I don't understand why this is tested: timeStamp should exist because it is initialized in src/main.cpp, right?